src/devices_models/devices/common/add_variable.jl

```diff
@@ -1,3 +1,134 @@
+struct AddVariableInputs
```

Comment:
How about we rename this VariableInputSpecification? Just an idea to improve naming.

Reply:
Would you do the same for making constraints and nodal expressions?

Reply:
Yes

Reply:
We would have
```
struct NodalExpressionInputSpecification
function make_nodal_expression_specification
struct DeviceRangeConstraintInputSpecification
function make_device_range_constraint_specification
```
I prefer `inputs` over `specification` but don't have a problem with it.  If @claytonpbarrows is onboard with you, I'll make the changes.

Reply:
Not putting a lot of thought into this I sort of prefer "inputs", but don't have a strong opinion.

Reply:
```
struct NodalExpressionSpecification
function make_nodal_expression_specification
struct DeviceRangeConstraintSpecification
function make_device_range_constraint_specification
```

or 

```
struct NodalExpressionInputInputs
function make_nodal_expression_inputs
struct DeviceRangeConstraintInputs
function make_device_range_constraint_inputs
```

Reply:
Per conversation on the phone and strong input from @jd-lara, go with `Spec`